### PR TITLE
Update gawk to v5.1.1 to fix CVE 2023-4156

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Update gawk to v5.1.1 to fix CVE 2023-4156

###### Change Log  

- Update gawk to v5.1.1 to fix CVE 2023-4156

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-4156

###### Test Methodology
- Buddy build pipeline successful build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=437581&view=results

### CODE REVIEW

**Review by mfrw — APPROVED** [review]
LGTM.
Since this is a toolchain package, I would request to verify with a full build before we merge.

**Comment by neha170** [comment]
> LGTM. Since this is a toolchain package, I would request to verify with a full build before we merge.

Thanks for highlighting this. Successfully built toolchain locally. Merging...

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Looks like a high CVE. Please rebase it against a fast-track branch.

File: toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh [hunk 1]
```diff
@@ -828,15 +828,15 @@ popd
 rm -rf diffutils-3.8
 touch /logs/status_diffutils_complete
 
-echo Gawk-5.1.0
-tar xf gawk-5.1.0.tar.xz
-pushd gawk-5.1.0
+echo Gawk-5.1.1
```
Comment (PawelWMS) [inline toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:831]: @anphel31, do we update the raw toolchain within one version of Mariner? I don't remember what's our policy here now, but I don't recall other toolchain changes updating this file. But maybe they were for packages, which were not part of the raw toolchain.

Comment (anphel31) [reply toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:831]: It's always a good idea to update the raw toolchain packages, but we have not been enforcing this in general in 2.0. Whether it's required depends on the nature of the CVE. If the CVE relates to a build-time issue with the tool, and a vulnerable version of the tool in the raw toolchain is used to build some other component, then the raw toolchain package should be updated.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**